### ISSUE TITLE
registration problems

### ISSUE THREAD

**Issue by LeeSmet** [itsyouonline/identityserver#598]
Some people report that they have issues with the registration flow in itsyouonline (in an oauth context):

- after registration - IYO immediately jumps back to the login-screen (via web browser)
- no email confirmation at all (email)
- same issues when using mobile phone

However this only seems to happen for a small portion of the users, and happens under unclear conditions (cannot be (easily) replicated it seems)

**Comment by svekeman** [comment]
All / we got more feedback - the problem with the bounce-back to the login might have to do with the fact that people have not registered their address or telephone number when using the (Threefold) website and/or app.

**Comment by LeeSmet** [comment]
After getting some feedback, the first point seems to be related to #602. For the second part, I checked the logs and no failure is reported on our end. I also haven't found anyone confirming this claim. although sometimes the mail seems to be received with a short delay. Since the api reports that is being send immediately, there is little we can do here AFAIC. As for the third part, the mobile and desktop website are the same with a slightly different layout and the same backend, so this is expected

**Comment by svekeman** [comment]
Thanks, on Wednesday the person with the issue is coming to our office. Let me see if I can replicate his issue somehow, if not, let's close this off...

**Comment by LeeSmet** [comment]
This morning, we identified the reported issue (first point) to be #604. Since this issue is more accurately tracked there, I'll close this for now

**Comment by robvanmieghem** [comment]
sine #604  is not solved, nor ws it in production, re-opening this issue

**Comment by LeeSmet** [comment]
After further investigation, the real issue turned out to be #606

**Comment by LeeSmet** [comment]
Moved to bugfix release milestone, fixed by f1fe736d08a53e2d249670215dc997071c65ad37

**Comment by robvanmieghem** [comment]
@LeeSmet where is the original issue  in the threefold app repo that resulted in the creation of this pretty generalissue after an initial investigation? It is not the intention that all problems there are reported here directly. Especially since passing empty scopes and stuff are obviously tf issues

**Comment by LeeSmet** [comment]
There is no issue pre dating this one in the threefold repo. But I understand what you are saying - in that sense I might have chosen a poor wording earlier as to how and why I created this issue here.

The problem was reported by the threefold team, along with info that several people managed to log in to the app without issue. Going on this info I assumed that it had to do with a browser/js issue.

Earlier today, before picking this up, Sabrina showed me the issue whilest regestering a new account as part of the TFA onboarding process. Since ahe had a similar device, this strenghtened my earlier assumption that it was a browser issue. 

So as it turns out, the TF guys somehow managed to put in the scopes when redirecting to log in and omit them when rederecting to register...

But from the data available, it wasn't clear that a lot of users already had an iyo account, thus going through login and providing all the scopes. Due to past issues with js browser support and the fact that the 2 reported issues happened on a similar/same device, I feared that this was a general issue and hence opened this issue here and not on the TFA repi
